staging/src/k8s.io/code-generator/cmd/applyconfiguration-gen/generators/types.go

```diff
@@ -19,18 +19,19 @@ package generators
 import "k8s.io/gengo/v2/types"
 
 var (
-	fmtSprintf           = types.Ref("fmt", "Sprintf")
-	syncOnce             = types.Ref("sync", "Once")
-	applyConfiguration   = types.Ref("k8s.io/apimachinery/pkg/runtime", "ApplyConfiguration")
-	groupVersionKind     = types.Ref("k8s.io/apimachinery/pkg/runtime/schema", "GroupVersionKind")
-	typeMeta             = types.Ref("k8s.io/apimachinery/pkg/apis/meta/v1", "TypeMeta")
-	objectMeta           = types.Ref("k8s.io/apimachinery/pkg/apis/meta/v1", "ObjectMeta")
-	rawExtension         = types.Ref("k8s.io/apimachinery/pkg/runtime", "RawExtension")
-	unknown              = types.Ref("k8s.io/apimachinery/pkg/runtime", "Unknown")
-	extractInto          = types.Ref("k8s.io/apimachinery/pkg/util/managedfields", "ExtractInto")
-	runtimeScheme        = types.Ref("k8s.io/apimachinery/pkg/runtime", "Scheme")
-	smdNewParser         = types.Ref("sigs.k8s.io/structured-merge-diff/v4/typed", "NewParser")
-	smdParser            = types.Ref("sigs.k8s.io/structured-merge-diff/v4/typed", "Parser")
-	testingTypeConverter = types.Ref("k8s.io/client-go/testing", "TypeConverter")
```

Comment:
is there anything we can add to make sure this doesn't creep back into non-test binaries? maybe in hack/verify-testing-import.sh

Reply:
Opened https://github.com/kubernetes/kubernetes/issues/131646 to track.  I'll try to get it assigned.